### PR TITLE
Another bunch of runtime fixes.

### PR DESCRIPTION
**Pull request by AnturK**
Explanation below where necessary.

### CODE REVIEW

File: code/game/objects/effects/effect_system/effect_system.dm [hunk 1]
```diff
@@ -70,7 +72,8 @@ would spawn and follow the beaker, even if it is carried or thrown.
 	for(var/j in 1 to steps_amt)
 		sleep(5)
 		step(E,direction)
-	addtimer(CALLBACK(src, .proc/decrement_total_effect), 20)
+	if(!QDELETED(src))
```
Comment (AnturK) [inline code/game/objects/effects/effect_system/effect_system.dm]: It's questionable if the elements should keep stepping when the source is deleted. Left as is for now.

File: code/modules/atmospherics/machinery/components/components_base.dm [hunk 2]
```diff
@@ -31,7 +31,7 @@
 	underlays.Cut()
 
 	var/turf/T = loc
-	if(level == 2 || !T.intact)
+	if(level == 2 || istype(T) && !T.intact)
```
Comment (AnturK) [inline code/modules/atmospherics/machinery/components/components_base.dm]: Pipe building uses temporary pipes in nullspace to get the connection directions. Very ugly, but there's no simple alternative.

File: code/modules/modular_computers/computers/item/computer_damage.dm [hunk 3]
```diff
@@ -23,6 +23,8 @@
 		new /obj/item/stack/sheet/metal(newloc, round(steel_sheet_cost/2))
 		for(var/C in all_components)
 			var/obj/item/computer_hardware/H = all_components[C]
+			if(QDELETED(H))
```
Comment (AnturK) [inline code/modules/modular_computers/computers/item/computer_damage.dm]: Once again explosion order issues.

File: code/modules/surgery/surgery.dm [hunk 4]
```diff
@@ -101,7 +101,7 @@
 		var/obj/item/tool = user.get_active_held_item()
 		if(S.try_op(user, target, user.zone_selected, tool, src, try_to_fail))
 			return TRUE
-		if(tool?.item_flags & SURGICAL_TOOL) //Just because you used the wrong tool it doesn't mean you meant to whack the patient with it
+		if(tool && tool.item_flags & SURGICAL_TOOL) //Just because you used the wrong tool it doesn't mean you meant to whack the patient with it
```
Comment (AnturK) [inline code/modules/surgery/surgery.dm]: There's possibly more of this class of errors, will have to search for procs that return obj/FALSE.

File: code/modules/atmospherics/machinery/components/components_base.dm (hunk 2 above)
Comment (MrPerson) [inline code/modules/atmospherics/machinery/components/components_base.dm]:
```suggestion
	if(level == 2 || (istype(T) && !T.intact))
```
Just makes the intent more clear that the && has higher priority than ||

File: code/modules/surgery/surgery.dm (hunk 4 above)
Comment (MrPerson) [reply code/modules/surgery/surgery.dm]: Seems like the kind of thing that should be linted for.

**Pull request merged**